### ISSUE TITLE
New local usage for Docker

### ISSUE THREAD

**Issue by vincent-zurczak** [roboconf/roboconf-platform#217]
Currently, two modes are planned with Roboconf.

* **Target**: instead of creating a VM, the DM creates a Docker container.
* **Plugin**: instead of running Puppet or Bash, the agent runs an application in a Docker container. See #200 

@Eriflow suggested another usage, which is in fact a variation of the second mode.
Instead of executing a script or a Puppet module, we could simply launch a Docker file with Docker. As an example, instead of running a command to launch NodeJS with a web application, the application would already contain a Docker file for the NodeJS component. It is a variation of the **plugin** approach in the sense where it is equivalent to having an agent running on the same machine than the DM.

We could have something to simplify this and having everything (DM + agent) running in the same Karaf. This will be most likely a documentation thing, as we already support such a use case today.

And the usage is a little bit specific because Roboconf would not be used to tests recipes but to ease complex development and deployment in Docker containers. That could bring new users too. Why not... And it does not seem hard to implement anyway. The real question is whether users will be ready to write their own Docker files. We will see if it seems realistic and usable after #200 is implemented.

**Comment by vincent-zurczak** [comment]
This issue is now a duplicate of #200.
